src/Microsoft.Extensions.Globalization.CultureInfoCache/CultureInfoCache.cs

```diff
@@ -18,15 +20,18 @@ public static partial class CultureInfoCache
         /// <see cref="KnownCultureNames"/> will be used.
         /// </summary>
         /// <param name="name">The culture name.</param>
+        /// <param name="supportedCultures">The cultures supported by the application.</param>
         /// <returns>
         /// A read-only cached <see cref="CultureInfo"/> or <c>null</c> a match wasn't found in
         /// <see cref="KnownCultureNames"/>.
         /// </returns>
-        public static CultureInfo GetCultureInfo(string name)
+        public static CultureInfo GetCultureInfo(string name, IList<CultureInfo> supportedCultures)
         {
+            var lowerInvariantName = name.ToLowerInvariant();
             // Allow only known culture names as this API is called with input from users (HTTP requests) and
             // creating CultureInfo objects is expensive and we don't want it to throw either.
-            if (name == null || !KnownCultureNames.Contains(name))
+            if (name == null || supportedCultures == null ||
+                !supportedCultures.Any(supportedCulture => supportedCulture.Name.ToLowerInvariant() == lowerInvariantName))
             {
```

Comment:
If we need to do a case-insensitive comparison, use a comparator rather than creating more strings